internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,224 @@
+			continue
+		}
+		confFilesPaths = append(confFilesPaths, confFileRelPath)
+	}
+	return confFilesPaths, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PHPDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return nil, nil, err
+	}
+	for _, de := range dirEntries {
+		if filepath.Ext(de.Name()) != phpExt {
+			continue
+		}
+		unnamedServices = []transformertypes.TransformerPlan{{
+			Mode:              t.Config.Spec.Mode,
+			ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			Paths: map[string][]string{
+				artifacts.ProjectPathPathType: {dir},
+			},
+		}}
+		return nil, unnamedServices, nil
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PHPDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var detectedPorts []int32
+		var phpConfig PhpTemplateConfig
+		var confFilePath string
+		confFiles, err := detectConfFiles(a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Debugf("Could not detect any conf files %s", err)
+		} else {
+			if len(confFiles) == 1 {
+				confFilePath = confFiles[0]
+			} else if len(confFiles) > 1 {
+				confFilePath = commonqa.GetConfFileForService(confFiles, a.Name)
+			}
+			if confFilePath != "" {
+				port := parseConfFile(filepath.Join(a.Paths[artifacts.ProjectPathPathType][0], confFilePath))
+				if port != 0 {
+					detectedPorts = append(detectedPorts, port)
+				}
+				phpConfig.ConfFile = confFilePath
+			}
+		}
+		if len(detectedPorts) == 0 {
+			detectedPorts = commonqa.GetPortsForService(detectedPorts, a.Name)
```

Comment:
I think depending on the service type, we should ask for one or more than one port. In this case, we want to expose only one port, that is exposed by the server.